enterprise/server/util/dynamic/dynamic.go

```diff
@@ -0,0 +1,80 @@
+package dynamic
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	digitsRegexp = regexp.MustCompile(`^\d+$`)
+)
+
+// FieldValues extracts values from an object given dot-separated field paths, returning
+// a mapping from each field path to the string representation of the value at that path.
+//
+// It is mostly useful when dealing with deeply nested structs parsed from a third-party
+// source.
+//
+// As paths are traversed, pointers are de-referenced (but only up to one level deep).
+// If a nil pointer is encountered the access path, an error is returned.
+//
+// Number-valued field names refer to slice indexes.
+func FieldValues(obj interface{}, fieldPaths ...string) (map[string]string, error) {
```

Comment:
I feel like there is probably a stupid hacky way to do this with a custom json parser but that's neither here nor there :p Working code FTW!

assuming this code came over from the old place as-is, LGTM.